lib/src/sync_worker_loop.dart

```diff
@@ -19,6 +19,8 @@ abstract class SyncWorkerConnection {
 }
 
 /// Persistent Bazel worker loop.
+///
+/// Extend this class and implement the `performRequest` method.
 abstract class SyncWorkerLoop {
```

Comment:
The analyzer_cli package currently does all its work synchronously, so the `SyncWorkerLoop` exists to support that. If you are open to converting it to be async, then we could probably remove the sync version.

However, there really isn't a lot of duplicated code here so I think its worth keeping support for both options.